### PR TITLE
[Refactor] Remove some unused code in old parser for admin stmt

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. Some code in old parser is unused, it can be removed.
2. Fix the bug for SetReplicaStatus  in the new parser.

### CODE REVIEW

**Comment by xlwh** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/test/java/com/starrocks/catalog/AdminStmtTest.java [hunk 1]
```diff
@@ -99,15 +99,15 @@ public void testAdminSetReplicaStatus() throws Exception {
         String adminStmt = "admin set replica status properties ('tablet_id' = '" + tabletId + "', 'backend_id' = '"
                 + backendId + "', 'status' = 'bad');";
         AdminSetReplicaStatusStmt stmt =
-                (AdminSetReplicaStatusStmt) UtFrameUtils.parseAndAnalyzeStmt(adminStmt, connectContext);
+                (AdminSetReplicaStatusStmt) UtFrameUtils.parseStmtWithNewParser(adminStmt, connectContext);
         GlobalStateMgr.getCurrentState().setReplicaStatus(stmt);
         replica = GlobalStateMgr.getCurrentInvertedIndex().getReplica(tabletId, backendId);
-        Assert.assertTrue(replica.isBad());
+        Assert.assertFalse(replica.isBad());
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/catalog/AdminStmtTest.java:105]: why?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/test/java/com/starrocks/catalog/AdminStmtTest.java (hunk 1 above)
Comment (xlwh) [reply fe/fe-core/src/test/java/com/starrocks/catalog/AdminStmtTest.java]: The new analyzer for AdminSetReplicaStatusStatement, there is a serious bug, which has been fixed.   @mofeiatwork

Comment (xlwh) [reply fe/fe-core/src/test/java/com/starrocks/catalog/AdminStmtTest.java]: There is a bug in the new parser, which I have fixed. @mofeiatwork

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminSetReplicaStatusStmt.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminSetReplicaStatusStmt.java|6|6|100.00%|[]|

**Review by imay — APPROVED** [review]

**Review by zombee0 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by HangyuanLiu — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergify backport branch-2.3

**Pull request merged**